samcli/commands/deploy/guided_context.py

```diff
@@ -119,8 +120,8 @@ def guided_prompts(self):
         Start an interactive cli prompt to collection information for deployment
 
         """
-        default_stack_name = self.stack_name
-        default_region = self.region or get_session().get_config_variable("region") or "us-east-1"
+        default_stack_name = self.stack_name or "sam-app"
```

Comment:
nit: We probably don't need to set the default value "sam-app" as I remember it is set somewhere before this step. But it shouldn't hurt to also have it here for now. My thought is that the default stack name is specific to which IaC platform is being used, so it should be set by the specific IaC plugin.